### DISCUSSION TITLE
Is there anyway we can pass custom jsonwebtoken config to Payload?

### DISCUSSION THREAD

**Discussion by rishi-afk** [payloadcms/payload#318] [Q&A]
I want to use asymmetric algorithm such as RSA with jwt for public-private key signing. Couldn't find anything in the docs. Any suggestions would be appreciated 🙏😅

**Comment by jmikrut** [comment]
Hey @rishi-afk — as of right now we use `passport-jwt` for everything JWT. 

We could probably expose options via Auth-enabled Collection configs for `passport-jwt`.

http://www.passportjs.org/packages/passport-jwt/

Do you see what you need here, in the `JWTStrategy` options? If so, we could try and inject custom collection auth options into the plugin within this file:

https://github.com/payloadcms/payload/blob/master/src/auth/strategies/jwt.ts#L14

What do you think?

**Reply by rishi-afk** [reply]
Hey @jmikrut thanks for the quick reply! 

Yes I just read it and we can provide an algorithm in the passport-jwt options. 

We are already injecting `secretOrKey` into this file
[https://github.com/payloadcms/payload/blob/master/src/auth/strategies/jwt.ts#L11](https://github.com/payloadcms/payload/blob/master/src/auth/strategies/jwt.ts#L11) 

So then we could also inject all the password-jwt options 
for a collection into the plugin within the same file.
